scripts/ironic/playbooks/ironic-pre-config.yml

```diff
@@ -0,0 +1,118 @@
+              bridge_waitport 10
+              bridge_fd 0
+              bridge_ports none
+              address 172.29.228.100
+              netmask 255.255.252.0
+              offload-sg off
+  
+          auto none
+          iface none inet manual
+          auto br-tftp
+          iface br-tftp inet static
+              bridge_stp off
+              bridge_waitport 10
+              bridge_fd 0
+              bridge_ports none
+              address 172.29.232.100
+              netmask 255.255.252.0
+              offload-sg off
+
+    - name: Start the ironic network interfaces
+      command: "ifup {{ item }}"
+      with_items:
+        - br-tftp
+        - br-ironic-ipmi
+
+    - name: Configure ironic network variables
+      config_template:
+        src: "/opt/rpc-openstack/openstack-ansible/etc/openstack_deploy/openstack_user_config.yml.aio"
+        dest: "/opt/rpc-openstack/openstack-ansible/etc/openstack_deploy/openstack_user_config.yml.aio"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_network_config }}"
+        config_type: "yaml"
+        list_extend: true
+
+    - name: Configure ironic OSA variables
+      config_template:
+        src: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_osa_config }}"
+        config_type: "yaml"
+        list_extend: true
+   
+    - name: Add additional configs to OSA variables
+      blockinfile:
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        block: |
+          neutron_dhcp_config:
+            dhcp-option-force: "26,1500"
+            dhcp-ignore: "tag:!known"
+            blog-facility: "/var/log/neutron/dnsmasq.log"
+          nova_scheduler_use_baremetal_filters: False
+          ironic_openstack_driver_list: "agent_ipmitool, fake"
+
+  vars:
+    ironic_network_config:
+      cidr_networks:
+        tftp: "172.29.232.0/22"
+        ironic-ipmi: "172.29.228.0/22"
+      used_ips:
+        - "172.29.232.1,172.29.232.50"
+        - "172.29.228.1,172.29.228.50"
+      global_overrides:
+        provider_networks:
+          - network:
+              container_bridge: "br-ironic-ipmi"
+              container_type: "veth"
+              container_interface: "eth_ipmi"
+              ip_from_q: "ironic-ipmi"
+              type: "raw"
+              group_binds:
+              - ironic-infra_hosts
+          - network:
+              container_bridge: "br-tftp"
+              container_type: "veth"
+              container_interface: "eth_tftp"
+              ip_from_q: "ironic-ipmi"
+              type: "flat"
+              net_name: "tftp"
+              ip_from_q: "tftp"
+              group_binds:
+                - neutron_linuxbridge_agent
+                - ironic_all
+    ironic_osa_config:
```

Comment:
Again, could we get these into `group_vars/ironic_all.yml` instead?

Reply:
we can consider this as a future iteration